src/Microsoft.Graph/Generated/Drives/Item/Items/Item/Invite/InvitePostRequestBody.cs

```diff
@@ -10,8 +10,8 @@ namespace Microsoft.Graph.Drives.Item.Items.Item.Invite {
     public class InvitePostRequestBody : IAdditionalDataHolder, IBackedModel, IParsable {
         /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
         public IDictionary<string, object> AdditionalData {
-            get { return BackingStore?.Get<IDictionary<string, object>>("additionalData"); }
-            set { BackingStore?.Set("additionalData", value); }
+            get { return BackingStore?.Get<IDictionary<string, object>>("AdditionalData"); }
```

Comment:
@baywet It looks like the change in https://github.com/microsoft/kiota/pull/343 is renaming the keys for the additionalData property in backingStore dictionaries. Although, this doesn't seem like it will cause any breaking side effects, does it make sense to rename the property? (This affects the other languages as well.)

https://github.com/microsoft/kiota/blob/d73545ea66f5e117c362c129df4a7f3af7743222/src/Kiota.Builder/KiotaBuilder.cs#L2146

cc @Ndiritu

Reply:
There might be breaking side effects with the current serialization experience in some languages where the deserialization map is checked case-sensitively. This would cause the properties to show up under additionalData instead of the model's properties. 

![image](https://github.com/microsoftgraph/msgraph-sdk-dotnet/assets/10958912/1d40bb90-72af-4a50-96e6-07ec8784998e)

Serialization would also produce upper-cased property names.

Reply:
It may be helpful to separate this into two aspects.
1. The `AdditionalData` property - this isn't the reference schema and is added by the Kiota builder. So the casing, naming is under our control.
2. Other properties in the changes (as in the screenshot) - These are changing due to the generator previously lowercasing the `WireName` property, which was not consistent with the reference schema. I think it would be correct to go ahead with the change. And any issue that arises should be fixed in the metadata.
![image](https://github.com/microsoftgraph/msgraph-sdk-dotnet/assets/6464005/b2814b59-e716-4dff-ba78-923b9040217d)

Reply:
1. To go further, I don't think the casing for the backing store matters as long as it's consistent for a single property between set and get.
2. yes, that was the point of the fix, make sure properties serialization names are not being normalized and match the casing the service is expecting/returning.

I'm not seeing anything abnormal with the changes mentioned in this conversation given the metadata we have.

Reply:
We could set the serialization name of the additional data property in the builder to conserve the casing we originally had as a precaution if you want.

Reply:
> We could set the serialization name of the additional data property in the builder to conserve the casing we originally had as a precaution if you want.

@baywet I agree that going forward with this shouldn't break anything and have no side effects. It's more around whether changing/keeping the casing is worth it as keeping it camelCased seems consistent in my opinion.

Reply:
the camel casing convention only applies to Microsoft graph. Other APIs might have other conventions (which was the reason for the fix). For the serialization name of additional properties, it's only going to be an opiniated choice.

Reply:
> For the serialization name of additional properties, it's only going to be an opiniated choice.

I hear you. Resolving this one for now.

Reply:
Thanks for clarifying @baywet. I also agree that we should let the schema determine what we generate.
I thought we might have a deserialization breaking change for Graph but had missed that the current changes only affect request bodies.